From: Simon McVittie <smcv@debian.org>
Date: Sun, 6 Oct 2019 16:15:01 +0100
Subject: build: Improve diagnostics when detecting Apache

This should help to debug what is going on if a distribution has Apache
installed in a layout that we don't understand.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://gitlab.gnome.org/GNOME/libsoup/merge_requests/103
---
 meson.build | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/meson.build b/meson.build
index dd30e66..5638d4e 100644
--- a/meson.build
+++ b/meson.build
@@ -233,10 +233,17 @@ if have_apache
     apache_modules_dirs = apache_modules_dirs_out.stdout().split(':')
     message('Apache modules directory: ' + apache_modules_dirs[0])
     cdata.set('APACHE_MODULE_DIR', apache_modules_dirs[0])
+    message('Apache SSL module directory: ' + apache_modules_dirs[1])
     cdata.set('APACHE_SSL_MODULE_DIR', apache_modules_dirs[1])
+    message('Apache PHP module file: ' + apache_modules_dirs[2])
     cdata.set('APACHE_PHP_MODULE_FILE', apache_modules_dirs[2])
+    message('Apache mod_unixd module directory: ' + (apache_modules_dirs[3] != '' ? apache_modules_dirs[3] : '(none)'))
     cdata.set('IF_HAVE_MOD_UNIXD', apache_modules_dirs[3] != '' ? '' : '#')
     cdata.set('HAVE_APACHE', have_apache)
+  else
+    message('Failed to locate necessary Apache modules for full test coverage')
+    message('stdout: ' + apache_modules_dirs_out.stdout())
+    message('stderr: ' + apache_modules_dirs_out.stderr())
   endif
 endif
 
